x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java

```diff
@@ -9,26 +9,33 @@
 
 import org.apache.lucene.util.Accountable;
 import org.apache.lucene.util.RamUsageEstimator;
+
+import java.time.Instant;
 import java.util.Objects;
 
 public class Ordinal implements Comparable<Ordinal>, Accountable {
 
     private static final long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(Ordinal.class);
 
-    private final long timestamp;
+    private final Object timestamp;
```

Comment:
I'm probably missing quite some context, but why not always using instant for holding the ts?

Reply:
Mostly for performance reasons, since timestamp handling is a core sequences operation and presumably most queries run on millis (as it had so far), i.e. longs.